cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedule.java

```diff
@@ -18,22 +18,37 @@
 
 
 /**
- * Defines a cron-based schedule for running a program. 
+ * Defines a cron-based schedule for running a program.
  */
 public class Schedule {
 
+  /**
+   * Defines different types of data schedules.
+   */
+  public enum DataType {
+    STREAM_SIZE,
```

Comment:
stream-size is not exactly a type of data. Stream is a data type, size is a trigger type. Since we only support size triggers for streams, perhaps you can just name this STREAM. And perhaps the enum should be called Source? Or Resource? @chtyim what do you think?